Tests/OpenCombineTests/PublisherTests/CountTests.swift

```diff
@@ -0,0 +1,169 @@
+    }
+
+    func testCountWaitsUntilFinishedToSend() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        let tracking = TrackingSubscriber(
+            receiveSubscription: { $0.request(.max(42)) }
+        )
+
+        countPublisher.subscribe(tracking)
+
+        _ = publisher.send(1)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        _ = publisher.send(2)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        _ = publisher.send(0)
+        XCTAssertEqual(tracking.history, [.subscription("Count")])
+
+        publisher.send(completion: .finished)
+        XCTAssertEqual(tracking.history, [.subscription("Count"),
+                                          .value(3),
+                                          .completion(.finished)])
+    }
+
+    func testDemand() {
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        var downstreamSubscription: Subscription?
+        let tracking = TrackingSubscriber(
+            receiveSubscription: {
+                $0.request(.max(42))
+                downstreamSubscription = $0
+            },
+            receiveValue: { _ in .max(4) }
+        )
+
+        countPublisher.subscribe(tracking)
+
+        XCTAssertNotNil(downstreamSubscription)
+        dump(type(of: downstreamSubscription!))
+
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+
+        XCTAssertEqual(publisher.send(0), .max(0))
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+
+        XCTAssertEqual(publisher.send(2), .max(0))
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+
+        downstreamSubscription?.request(.max(95))
+        downstreamSubscription?.request(.max(5))
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+
+        downstreamSubscription?.cancel()
+        downstreamSubscription?.cancel()
+        XCTAssertEqual(subscription.history, [.requested(.unlimited),
+                                              .cancelled])
+
+        downstreamSubscription?.request(.max(50))
+        XCTAssertEqual(subscription.history, [.requested(.unlimited),
+                                              .cancelled])
+    }
+
+    func testAddingSubscriberRequestsUnlimitedDemand() {
+        // When
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let countPublisher = publisher.count()
+        let tracking = TrackingSubscriber()
+
+        // Given
+        XCTAssertEqual(subscription.history, [])
+        countPublisher.subscribe(tracking)
+
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+    }
+
+    func testNoDemand() {
```

Comment:
If I'm not mistaken, the two above tests cover the same cases as this test. Maybe remove it?

Reply:
I'm testing that the "receiveSubscription" and "receiveValue" have different cases in them, so they cover similar ground but are ensuring that there are no differences between different demand requests.  Feel free to add a commit removing either one if you think that's overkill.